h/static/scripts/controllers/lozenge-controller.js

```diff
@@ -0,0 +1,28 @@
+'use strict';
+
+var Controller = require('../base/controller');
+var LozengeHelper = require('../util/lozenge-helper');
+
+class LozengeController extends Controller {
+  constructor(el) {
+    super(el);
+    el.querySelector('.lozenge-close').addEventListener('mousedown', () => {
+      // remove the DOM element
+      el.remove();
+      // update input field
+      this._inputHidden = document.querySelector('.js-search-bar').querySelector('.search-bar__input-hidden');
+      var queryTerms = LozengeHelper.getQueryTerms(this._inputHidden.value);
+
+      queryTerms = queryTerms.map(function(term) {
+        if (term !== el.querySelector('.lozenge-content').textContent) {
+          return term;
+        }
+      });
+      this._inputHidden.value = queryTerms.join(' ');
+      // submit form
+      document.querySelector('.js-search-bar').querySelector('form').submit();
```

Comment:
We need to let the search controller know that we have deleted a lozenge but we don't need to directly invoke the submit of the search bar or update the hidden input valu as these are concerns of the searchbarcontroller.